src/apprunner/commands/pauseService.ts

```diff
@@ -7,36 +7,26 @@ import * as nls from 'vscode-nls'
 const localize = nls.loadMessageBundle()
 
 import * as vscode from 'vscode'
-import * as telemetry from '../../shared/telemetry/telemetry'
 import * as localizedText from '../../shared/localizedText'
 import { showConfirmationMessage } from '../../shared/utilities/messages'
 import { AppRunnerServiceNode } from '../explorer/apprunnerServiceNode'
 import { PromptSettings } from '../../shared/settings'
+import { getTelemetryLogger } from '../../shared/telemetry/recorder'
 
 export async function pauseService(node: AppRunnerServiceNode): Promise<void> {
-    let telemetryResult: telemetry.Result = 'Failed'
+    const prompts = PromptSettings.instance
+    const shouldNotify = await prompts.isPromptEnabled('apprunnerNotifyPause')
+    const notifyPrompt = localize(
+        'aws.apprunner.pauseService.notify',
+        'Your service will be unavailable while paused. ' +
+            'You can resume the service once the pause operation is complete.'
+    )
+    const confirmationOptions = { prompt: notifyPrompt, confirm: localizedText.ok, cancel: localizedText.cancel }
 
-    try {
-        const prompts = PromptSettings.instance
-        const shouldNotify = await prompts.isPromptEnabled('apprunnerNotifyPause')
-        const notifyPrompt = localize(
-            'aws.apprunner.pauseService.notify',
-            'Your service will be unavailable while paused. ' +
-                'You can resume the service once the pause operation is complete.'
-        )
-        const confirmationOptions = { prompt: notifyPrompt, confirm: localizedText.ok, cancel: localizedText.cancel }
-
-        if (shouldNotify && !(await showConfirmationMessage(confirmationOptions, vscode.window))) {
-            telemetryResult = 'Cancelled'
-            return
-        }
-
-        await node.pause()
-        telemetryResult = 'Succeeded'
-    } finally {
-        telemetry.recordApprunnerPauseService({
-            result: telemetryResult,
-            passive: false,
-        })
+    if (shouldNotify && !(await showConfirmationMessage(confirmationOptions, vscode.window))) {
+        getTelemetryLogger('ApprunnerPauseService').recordResult('Cancelled')
```

Comment:
Ideally `getTelemetryLogger` would just let us record any metadata we want. But that's not easy with the current codegen.

I've also now created the problem of duplicate entries. Still thinking about the best way to handle multiple functions trying to record the same metadata. Right now it just uses the most recent.